cirq/google/optimizers/convert_to_sycamore_gates.py

```diff
@@ -43,14 +45,19 @@ class ConvertToSycamoreGates(circuits.PointOptimizer):
         Otherwise raises a TypeError.
     """
 
-    def __init__(self, ignore_failures=False) -> None:
+    def __init__(self,
+                 tabulation: Optional[GateTabulation] = None,
+                 ignore_failures=False) -> None:
         """
         Args:
+            tabulation: If set, a tabulation for the Sycamore gate to use for
+                decomposing gates.
             ignore_failures: If set, gates that fail to convert are forwarded
                 unchanged. If not set, conversion failures raise a TypeError.
         """
         super().__init__()
         self.ignore_failures = ignore_failures
+        self.tabulation = tabulation
```

Comment:
I would do some validation here, like checking that tabulation is None or actually a valid `GateTabulation`. Otherwise the user gets a more mysterious "tabulation object has no compile_two_qubit_gate" method

Reply:
Done - but humor me here, as I'm not great at Python. Isn't that was mypy is supposed to do?

Reply:
mypy is basically a linter. The type annotations do nothing at runtime

Reply:
Sure, but in theory, it would prevent this situation from ever happening, because we wouldn't be able to submit any code that violates the types. Coming from a TypeScript background, the type system doesn't do anything at runtime but can be relied on to prevent these sort of situations because it runs on every compile. I guess the difference here is that mypy is not run as often, e.g. during development.

Reply:
Bear in mind that cirq is basically an SDK and users will be calling this function from their hacky scripts. This type checking and my feedback about documenting "where you get a tabulation object" are two sides of the same coin: we've got to help the users use this new feature!

Reply:
Sounds reasonable to me, thanks for the guidance!